pkg/controller/applicationlayer/applicationlayer_controller.go

```diff
@@ -277,12 +308,71 @@ func updateApplicationLayerWithDefaults(al *operatorv1.ApplicationLayer) {
 
 // validateApplicationLayer validates ApplicationLayer
 func validateApplicationLayer(al *operatorv1.ApplicationLayer) error {
-	lcSpec := al.Spec.LogCollection
 
-	// If ApplicationLayer spec exists then LogCollection should be set.
-	// TODO: when we will have multiple features in future this should change to at least one feature being set
-	if lcSpec == nil {
-		return fmt.Errorf("missing required LogCollection spec in applicationLayer")
+	// If ApplicationLayer spec exists then one of its features should be set.
+	if al.Spec.LogCollection == nil && al.Spec.WebApplicationFirewall == nil {
+		return fmt.Errorf("ApplicationLayer is not configured")
```

Comment:
This is a user facing error, it needs to be more precise about what the problem is and how it can be resolved.

e.g., "At least one of webApplicationFirewall or locCollector must be specified on ApplicationLayer resource"

Reply:
done.